spec/unit/policyfile_spec.rb

```diff
@@ -0,0 +1,145 @@
+# frozen_string_literal: true
+
+require 'git'
+require 'spec_helper'
+
+RSpec.describe PolicyChangelog do
+  let(:pf_dir) do
+    File.join(File.dirname(__FILE__), '../data')
+  end
+
+  let(:changelog) do
+    PolicyChangelog.new('users', File.join(pf_dir, 'Policyfile.rb'))
+  end
+
+  describe '#versions' do
+    context 'when type is current' do
+      it 'returns correct current versions' do
+        locks = JSON.parse(File.read(File.join(pf_dir, 'pf_lock_current.json')))['cookbook_locks']
+        current_versions = {
+          'sudo' => { 'current_version' => '3.5.0' },
+          'users' => { 'current_version' => '4.0.0' }
+        }
+        expect(changelog.versions(locks, 'current')).to eq(current_versions)
+      end
+    end
+
+    context 'when type is target' do
+      it 'returns correct target versions' do
+        locks = JSON.parse(File.read(File.join(pf_dir, 'pf_lock_target.json')))['cookbook_locks']
+        target_versions = {
+          'sudo' => { 'target_version' => '3.5.0' },
+          'users' => { 'target_version' => '5.3.1' }
+        }
+        expect(changelog.versions(locks, 'target')).to eq(target_versions)
+      end
+    end
+
+    context 'when type is not current nor target' do
+      it 'raises and exception' do
+        expect { changelog.versions(nil, 'toto') }.to raise_error(RuntimeError)
+      end
+    end
```

Comment:
OPT: add test for when locks is empty? (an easy one just for completeness but really not necessary)

Reply:
Added the test